src/lib/storage/dictionary_segment.cpp

```diff
@@ -209,7 +207,26 @@ std::shared_ptr<const BaseCompressedVector> DictionarySegment<T>::attribute_vect
 
 template <typename T>
 ValueID DictionarySegment<T>::null_value_id() const {
-  return ValueID{static_cast<ValueID::base_type>(_dictionary_span->size())};
+  return ValueID{static_cast<ValueID::base_type>(_dictionary->size())};
+}
+
+template <typename T>
+void DictionarySegment<T>::serialize(std::ofstream& ofstream) const {
+  /*
+   * For a description of how dictionary segments look, see the following PR:
+   *    https://github.com/hyrise-mp-22-23/hyrise/pull/94
+   */
+  const auto compressed_vector_type_id =
+      StorageManager::resolve_persisted_segment_encoding_type_from_compression_type(compressed_vector_type().value());
+  StorageManager::export_value(static_cast<uint32_t>(compressed_vector_type_id), ofstream);
+
+  // Ee need to ensure that every part can be mapped with a uint32_t map.
+  StorageManager::export_value(static_cast<uint32_t>(dictionary()->size()), ofstream);
```

Comment:
Not to be addressed: those methods should not be part of the stroage manager but rather some util class.